pymc_marketing/mmm/models/components/saturation.py

```diff
@@ -0,0 +1,334 @@
+from typing import Dict, Optional, Union
+
+import pymc as pm
+
+from pymc_marketing.mmm.transformers import (
+    hill_saturation,
+    logistic_saturation,
+    michaelis_menten,
+)
+from pymc_marketing.mmm.utils import _get_distribution, _validate_model_config
+
+
+### SATURATION FUNCTIONS
+class BaseFunction:
+    def __init__(self, max_lagging, model):
+        self.max_lagging = max_lagging
+        self.model = model
```

Comment:
same recommendation here. 

Also think that these will be very similar so potentially combining in a components.base module

Reply:
They might deviate a bit in order to suppor the experimentation integration. 

For all of these, the `add_lift_test_measurements` would just need something like this in order to be supported: https://github.com/wd60622/pymc-marketing/blob/lift-test-calibration/pymc_marketing/mmm/lift_test.py#L372-L378

However we can do that after both are merged